client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/ClusterClientEventStressTest.java

```diff
@@ -0,0 +1,285 @@
+
+@Test(groups = "stress", testName = "client.hotrod.event.ClusterClientEventStressTest")
+public class ClusterClientEventStressTest extends MultiHotRodServersTest {
+
+   private static final Log log = LogFactory.getLog(ClusterClientEventStressTest.class);
+
+   static final int NUM_SERVERS = 3;
+   static final int NUM_OWNERS = 2;
+
+   static final int NUM_CLIENTS = 1;
+   static final int NUM_THREADS_PER_CLIENT = 6;
+//   static final int NUM_THREADS_PER_CLIENT = 36;
+
+   static final int NUM_OPERATIONS = 1_000; // per thread, per client
+//   static final int NUM_OPERATIONS = 300; // per thread, per client
+//   static final int NUM_OPERATIONS = 600; // per thread, per client
+
+   static final int NUM_EVENTS = NUM_OPERATIONS * NUM_THREADS_PER_CLIENT * NUM_CLIENTS * NUM_CLIENTS;
+
+//   public static final int NUM_STORES = NUM_OPERATIONS * NUM_CLIENTS * NUM_THREADS_PER_CLIENT;
+//   public static final int NUM_STORES_PER_SERVER = NUM_STORES / NUM_SERVERS;
+//   public static final int NUM_ENTRIES_PER_SERVER = (NUM_STORES * NUM_OWNERS) / NUM_SERVERS;
+
+   static Set<String> ALL_KEYS = new ConcurrentHashSet<>();
+
+   static ExecutorService EXEC = Executors.newCachedThreadPool();
+
+   static ClientEntryListener listener;
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      createHotRodServers(NUM_SERVERS, getCacheConfiguration());
+   }
+
+   private ConfigurationBuilder getCacheConfiguration() {
+      ConfigurationBuilder builder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false);
+      builder.clustering().hash().numOwners(NUM_OWNERS)
+//            .expiration().lifespan(1000).maxIdle(1000).wakeUpInterval(5000)
+            .expiration().maxIdle(1000).wakeUpInterval(5000)
+            .jmxStatistics().enable();
+      return hotRodCacheConfiguration(builder);
+   }
+
+   RemoteCacheManager getRemoteCacheManager(int port) {
+      org.infinispan.client.hotrod.configuration.ConfigurationBuilder builder =
+            new org.infinispan.client.hotrod.configuration.ConfigurationBuilder();
+      builder.addServer().host("127.0.0.1").port(port);
+      RemoteCacheManager rcm = new InternalRemoteCacheManager(builder.build());
+      rcm.getCache();
+      return rcm;
+   }
+
+   Map<String, RemoteCacheManager> createClients() {
+      Map<String, RemoteCacheManager> remotecms = new HashMap<>(NUM_CLIENTS);
+      for (int i = 0; i < NUM_CLIENTS; i++)
+         remotecms.put("c" + i, getRemoteCacheManager(server(0).getPort()));
+
+      return remotecms;
+   }
+
+   public void testAddClientListenerDuringOperations() {
+      CyclicBarrier barrier = new CyclicBarrier((NUM_CLIENTS * NUM_THREADS_PER_CLIENT) + 1);
+      List<Future<Void>> futures = new ArrayList<>(NUM_CLIENTS * NUM_THREADS_PER_CLIENT);
+      List<ClientEntryListener> listeners = new ArrayList<>(NUM_CLIENTS);
+      Map<String, RemoteCacheManager> remotecms = createClients();
+
+//      assertStatsBefore(remotecms);
```

Comment:
If I added the listeners at the start, I could make some calculations and assertions, but in the test I actually add the listener midway, so can't make such assertion on number of events. I left these commented in case I want to try out again to add listener at the start at some point.